internal/storage/activities/delete_amss.go

```diff
@@ -72,17 +72,18 @@ func (a *DeleteFromAMSSLocationActivity) Execute(
 	}
 
 	// Set up ticker channel if not provided.
-	if a.tickerCh == nil {
+	tickerCh := a.tickerCh
```

Comment:
I think that even with this change, the current functionality will cause concurrent executions of the activity to drain the same channel which will cause two executions to tick at half the ticker interval, see:
https://go.dev/play/p/ab_1291pDTj

If you look at the timings from that example the "activity 1" executions are alternating receives from the ticker channel.

Is there a reason to inject a ticker channel into the activity rather than just passing a `time.Duration` interval, and then starting the ticker inside `Execute()`?

Reply:
In that example you are manipulating the ticket channel directly in the activity struct. The ticket can only be provided using a different constructor, intended for tests:

https://github.com/artefactual-sdps/enduro/blob/dev/issue-1401-feedback/internal/storage/activities/delete_amss.go#L36-L46

Production code will use the other constructor:

https://github.com/artefactual-sdps/enduro/blob/dev/issue-1401-feedback/cmd/enduro/main.go#L532

In this case each activity will create its own channel in a local var. Previously I was updating the activity struct, even if the channel was not provided.

Reply:
I think there are a few advantages to passing a poll interval in the activity constructor, rather than passing a `<-chan time.Time`:

1. No need for a separate constructor and code path for tests
2. Easier to understand
3. Makes it easier to set the poll interval by configuration, if we want to do that in the future

I think in general it's good practice to try and minimize the differences between the code run in production and in test.